lib/Socket.js

```diff
@@ -40,16 +40,16 @@ class Socket extends EventEmitter {
 
     const originalEmit = this.emit;
     this.emit = function () {
-      log.debug('[Socket:emit]', arguments[0], arguments[1]);
+      debug('[emit] %O %O', arguments[0], arguments[1]);
       originalEmit.apply(this, arguments);
     };
 
     this.connect({address})
       .then(() => {
         this.authenticate({password});
-      })
-      .catch(err => {
-        log.error('Connection or Authentication failed.', err);
+      }, err => {
```

Comment:
Oh actually, let me test that real quick before you merge. I'm not 100% sure that will work as I think.